src/library/EndpointManagement/EndpointManagementClient.cs

```diff
@@ -454,6 +398,20 @@ private async Task EnsureEndpointManagerRunningAsync(CancellationToken cancellat
                                                 cancellationToken: cancellationToken,
                                                 out string outPut);
 
+                    //  exit code not equal to zero occurs when pkexec doesn't have root access and have to retry with sudo
+                    if (launchExitCode != 0 && _platform.IsLinux && !_environmentVariables.IsCodespaces) {
+                         _log.Info($"pkexec failed with exitCode {launchExitCode}, retrying with sudo");
+                        (fileName, command) = await GetEndpointManagerLaunchArguments(currentUserName, logFileDirectory, cancellationToken);
+                        fileName = "sudo"; // replace pkexec with sudo

```

Comment:
if user doesn't have sudo access, it would still fail but atleast this is another try for users for who had pkexec failed but had sudo access. For non-sudo user I will look into see how to handle this further. Thank you.